src_common/network.c

```diff
@@ -66,73 +66,33 @@ static const network_info_t NETWORK_MAPPING[] = {
     {.chain_id = 39797, .name = "Energi", .ticker = "NRG"},
     {.chain_id = 248, .name = "Oasys", .ticker = "OAS"}};
 
-uint64_t get_tx_chain_id(void) {
-    uint64_t chain_id = 0;
-
-    switch (txContext.txType) {
-        case LEGACY:
-            chain_id = u64_from_BE(txContext.content->v, txContext.content->vLength);
-            break;
-        case EIP2930:
-        case EIP1559:
-            chain_id = u64_from_BE(tmpContent.txContent.chainID.value,
-                                   tmpContent.txContent.chainID.length);
-            break;
-        default:
-            PRINTF("Txtype `%d` not supported while generating chainID\n", txContext.txType);
-            break;
-    }
-    return chain_id;
-}
-
-uint64_t get_app_chain_id(void) {
-    return chainConfig->chainId;
-}
-
-static uint64_t get_chain_id(e_net_type type) {
-    return (type == APP) ? get_app_chain_id() : get_tx_chain_id();
-}
-
-static const network_info_t *get_network(e_net_type type) {
-    uint64_t chain_id = get_chain_id(type);
-    for (size_t i = 0; i < sizeof(NETWORK_MAPPING) / sizeof(*NETWORK_MAPPING); i++) {
-        if (NETWORK_MAPPING[i].chain_id == chain_id) {
-            return (const network_info_t *) PIC(&NETWORK_MAPPING[i]);
+static const network_info_t *get_network_from_chain_id(const uint64_t *chain_id) {
+    for (size_t i = 0; i < ARRAYLEN(NETWORK_MAPPING); i++) {
+        if (NETWORK_MAPPING[i].chain_id == *chain_id) {
+            return (const network_info_t *) &NETWORK_MAPPING[i];
```

Comment:
PIC in this case ?

Reply:
`PIC` is only necessary for accessing the elements of the structs.